inc/deployable.class.php

```diff
@@ -0,0 +1,188 @@
+         exit(0);
+      }
+
+      // Download range defaults to the full file
+      // get file metadata
+      $size = $fields['filesize'];
+      $begin = 0;
+      $end = $size - 1;
+      $mimeType = 'application/octet-stream';
+      $time = date('r', filemtime($streamSource));
+
+      // Open the file
+      $fileHandle = @fopen($streamSource, 'rb');
+      if (!$fileHandle) {
+         header("HTTP/1.0 500 Internal Server Error");
+         exit(0);
+      }
+
+      // set range if specified by the client
+      if (isset($_SERVER['HTTP_RANGE'])) {
+         $matches = null;
+         if (preg_match('/bytes=\h*(\d+)?-(\d*)[\D.*]?/i', $_SERVER['HTTP_RANGE'], $matches)) {
+            if (!empty($matches[1])) {
+               $begin = intval($matches[1]);
+            }
+            if (!empty($matches[2])) {
+               $end = min(intval($matches[2]), $end);
+            }
+         }
+      }
+
+      // seek to the begining of the range
+      $currentPosition = $begin;
+      if (fseek($fileHandle, $begin, SEEK_SET) < 0) {
+         header("HTTP/1.0 500 Internal Server Error");
+         exit(0);
+      }
+
+      // send headers to ensure the client is able to detect a corrupted download
+      // example : less bytes than the expected range
+      // send meta data
+      // setup client's cache behavior
+      header("Expires: Mon, 26 Nov 1962 00:00:00 GMT");
+      header('Pragma: private'); /// IE BUG + SSL
+      header('Cache-control: private, must-revalidate'); /// IE BUG + SSL
+      header("Content-disposition: attachment; filename=\"" . $filename . "\"");
+      header("Content-type: $mimeType");
+      header("Last-Modified: $time");
+      header('Accept-Ranges: bytes');
+      header('Content-Length:' . ($end - $begin + 1));
+      header("Content-Range: bytes $begin-$end/$size");
+      header("Content-Transfer-Encoding: binary\n");
+      header('Connection: close');
+
+      $httpStatus = 'HTTP/1.0 200 OK';
+      if ($begin > 0 || $end < $size - 1) {
+         $httpStatus = 'HTTP/1.0 206 Partial Content';
+      }
+      header($httpStatus);
+
+      // Sends bytes until the end of the range or connection closed
+      while (!feof($fileHandle) && $currentPosition < $end && (connection_status() == 0)) {
+         // allow a few seconds to send a few KB.
+         set_time_limit(10);
+         $content = fread($fileHandle, min(1024 * 16, $end - $currentPosition + 1));
+         if ($content === false) {
+            header("HTTP/1.0 500 Internal Server Error", true); // Replace previously sent headers
+            exit(0);
+         }
+         print $content;
+         flush();
+         $currentPosition += 1024 * 16;
+      }
+
+      // Endnow to prevent any unwanted bytes
+      exit(0);
+   }
+
+   /**
+    * @param string $itemtype
+    * @param integer $itemId
+    * @param PluginFlyvemdmTask $task
+    */
+   protected function publishTaskAction($itemtype, $itemId, PluginFlyvemdmTask $task) {
```

Comment:
I suggest to rename this method post_updateItem() and reuse the calls to PluginFlyvemdmTask::publishPolicy() and  PluginFlyvemdmTask::createTaskStatuses() for  readability.

In the 2 inherited classes, just call parent::post_updateItem()

Reply:
post_updateItem must be compatible with its parent method on commonDBTM so I can't use that name with those arguments also the child classes have different logic implementations. 

This method now is a little more isolated and testable.